### PR TITLE
test(web): expand utility coverage

### PR DESCRIPTION
**Pull request by Peyton-Spencer**
## Summary
- Add focused coverage for `src/web/agent-channels.ts` and `src/slack-jid.ts`, including escaping, fallback naming, context attributes, and scoped Slack JID parsing.
- Expand deterministic utility coverage for `src/resume-position-store.ts` and `src/web/ipc-events.ts`, including sanitization, persistence failures, ring-buffer eviction, and fixed-time event timestamps.
- Raise the suite from 998 passing tests across 53 files to 1007 passing tests across 55 files, while documenting the next unpaired modules to target.

## Verification
- `bun test`

## Coverage Notes
- Files covered in this PR: `src/web/agent-channels.ts`, `src/slack-jid.ts`, deeper branch coverage for `src/resume-position-store.ts`, and deeper branch coverage for `src/web/ipc-events.ts`.
- Remaining paired-file coverage gaps still worth targeting: `src/backends/local-backend.ts`, `src/backends/local-backend.isolated.ts`, `src/discovery/peer-client.ts`, `src/discovery/trust-store.ts`, `src/ipc.ts`, `src/web/context-files.ts`, `src/web/context-viewer.ts`, `src/web/dashboard.ts`, `src/web/network.ts`, `src/web/page-scripts.ts`, `src/web/shared.ts`, and `src/whatsapp-auth.ts`.

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for resume position storage with in-memory and persistent state handling.
  * Added new test suite for Slack JID parsing validation.
  * Introduced comprehensive tests for agent-channel rendering logic, including channel enrichment and display features.
  * Enhanced event buffering tests with deterministic timing framework for improved reliability.

### CODE REVIEW

File: src/web/agent-channels.test.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    expect(contextHtml).toContain('data-select-channel');
+    expect(contextHtml).toContain('data-folder="agent-one"');
+    expect(contextHtml).toContain('data-server-folder="servers/root"');
+    expect(contextHtml).toContain(
+      'data-agent-context-folder="agents/agent-one"',
+    );
+    expect(contextHtml).toContain('data-channel-folder="servers/root/general"');
+    expect(contextHtml).toContain('data-category-folder="servers/root"');
+  });
+});
```
Comment (Peyton-Spencer) [inline src/web/agent-channels.test.ts]: **Suggestion:** Good coverage on `renderAgentGroups` — escaping and context-attribute tests are exactly right. One gap worth adding: test with an agent that has *no* channels (empty subscription map) to confirm the renderer handles the empty-list gracefully rather than omitting the agent card entirely.

File: src/slack-jid.test.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  it('allows channel ids with separators after the second segment', () => {
+    expect(parseScopedSlackJid('slack:OPS:thread-abc:1700000000.000100')).toEqual(
+      {
+        botId: 'OPS',
+        channelId: 'thread-abc:1700000000.000100',
+      },
+    );
+  });
+
+  it('rejects unscoped or malformed JIDs', () => {
```
Comment (Peyton-Spencer) [inline src/slack-jid.test.ts]: **Suggestion:** Nice rejection coverage. Consider also testing an empty string (`parseScopedSlackJid('')`) — common edge case for JID parsers.

**Review by Peyton-Spencer — COMMENTED** [review]
**Review from @Clayton** — Good coverage additions (998→1007). CI is red on **format** and **typecheck** — needs a fix push before merge. Details in inline comments.

**Summary:**

- ✅ `src/web/agent-channels.test.ts` — Thorough. Escaping, multi-agent filtering, context attribute toggles all covered well.
- ✅ `src/slack-jid.test.ts` — Clean and complete for the surface area (valid, multi-segment, invalid cases).
- ✅ `src/web/ipc-events.test.ts` — Nice refactor toward deterministic timestamps with the FixedDate approach. Ring buffer eviction is tested more explicitly now.
- ⚠️ `src/resume-position-store.test.ts` — Good expansion with describe blocks and logger spies, but the mock adapter `read()` returns cause 5 typecheck errors that block CI.
- 🔴 **Prettier** fails on `src/slack-jid.test.ts` and `src/web/agent-channels.test.ts` — run `bunx prettier --write` on both.

**Review by Peyton-Spencer — COMMENTED** [review]
Solid test additions — the `agent-channels.test.ts` coverage is particularly thorough (escaping, platform icons, context attrs, agent filtering). The `FixedDate` monkey-patch in ipc-events is a nice deterministic improvement over the old timestamp assertions.

Two CI blockers need fixing before merge:

1. **Typecheck (5 errors)** — the mock `PersistentStateAdapter.read` implementations return concrete types that don't satisfy the generic `<T>` signature. See inline comment for fix.
2. **Prettier (2 files)** — `src/slack-jid.test.ts` and `src/web/agent-channels.test.ts` need formatting.

Minor suggestions inline. Tests themselves are well-structured and assertions are meaningful.

**Review by Peyton-Spencer — COMMENTED** [review]
Triage: the scope is clean for a coverage-only follow-up, but it is not merge-ready yet because CI is red (`format`, `typecheck`). Once those are green this looks like a straightforward test-only merge.
